gtda/metaestimators/collection_transformer.py

```diff
@@ -0,0 +1,140 @@
+"""CollectionTransformer meta-estimator."""
+# License: GNU AGPLv3
+
+from functools import reduce
+from operator import and_
+
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.base import clone
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+from gtda.utils import check_collection
+
+
+class CollectionTransformer(BaseEstimator, TransformerMixin):
+    """Meta-transformer for applying a fit-transformer to each input in a
+    collection.
+
+    If `transformer` possesses a ``fit_transform`` method,
+    ``CollectionTransformer(transformer)`` also possesses a
+    :meth:`fit_transform` method which, on each entry in its input ``X``,
+    fit-transforms a clone of `transformer`. A collection (list or ndarray) of
+    outputs is returned.
+
+    Parameters
+    ----------
+    transformer : object
+        The fit-transformer instance from which the transformer acting on
+        collections is built. Should implement ``fit_transform``.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use in a joblib-parallel application of
+        `transformer`'s ``fit_transform`` to each input. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using
+        all processors.
+
+    parallel_backend_prefer :  ``"processes"`` | ``"threads"`` | ``None``, \
```

Comment:
I am really not sure why this should be a parameter. IMO, Joblib backend should be selected as a global parameter. Can you explain this choice?